test/parallel/test-windows-failed-heap-allocation.js

```diff
@@ -0,0 +1,22 @@
+'use strict';
+const common = require('../common');
+
+// This test ensures that an out of memory error exits with code 134 on Windows
+
+if (!common.isWindows) return common.skip('Windows-only');
+
+const assert = require('assert');
+const { exec } = require('child_process');
+
+if (process.argv[2] === 'heapBomb') {
+  // heap bomb, imitates a memory leak quickly
+  const fn = (nM) => [...Array(nM)].map((i) => fn(nM ** 2));
+  fn(2);
+}
+
+// --max-old-space-size=3 is the min 'old space' in v8, explodes fast
+const cmd = `"${process.argv[0]}" --max-old-space-size=3 "${process.argv[1]}"`;
+exec(`${cmd} heapBomb`, (err) => {
+  const msg = `Wrong exit code of ${err.code}! Expected 134 for abort`;
+  assert.strictEqual(err.code, 134, msg);
```

Comment:
Might be worth a short comment explaining why `common.nodeProcessAborted()` is not used.